src/mechanics/main_menu_buttons.rs

```diff
@@ -0,0 +1,27 @@
+use bevy::prelude::*;
+use bevy::app::AppExit;
+
+use crate::AppState;
+use crate::visuals::main_menu::MainMenuButtonTypes;
+
+pub fn button_system(
+    mut exit_event: EventWriter<AppExit>,
+    mut next_state: ResMut<NextState<AppState>>,
+    mut interaction_query: Query<(&Interaction, &MainMenuButtonTypes), (Changed<Interaction>, With<Button>)>
+) {
+    for (interaction, button_type) in &mut interaction_query {
+        match *interaction {
+            Interaction::Clicked => {
+                match button_type {
+                    MainMenuButtonTypes::Play => next_state.set(AppState::InGame),
+                    MainMenuButtonTypes::Quit => exit_event.send(AppExit)
+                }
+            }
+            Interaction::Hovered => {
+                //println! ("Hover");
+            }
+            Interaction::None => {
+            }
```

Comment:
Since you're only checking on whether a button has been clicked, what do you think about changing this section of code to use an if statement instead?

```rust
if *interaction != InteractionClicked => {
    continue;
}

match button_type {
    MainMenuButtonTypes::Play => next_state.set(AppState::InGame),
    MainMenuButtonTypes::Quit => exit_event.send(AppExit)
}
```

Reply:
I'm a bit torn on this. On the one hand, we'll have to change it back once we add behaviors for the other two arms. On the other, it's not hard to switch it back to a match statement when we get to the polish phase. I'll change it for now, in the interest of only keeping the code that's currently being used.